#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 50
}
Viewpoint {
  orientation 0 1 0 0
  position 5.6439099717598515e-12 1.5593044161401617e-13 16.336276974623832
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 1 0 0 1.57
  floorSize 4.5 6
  wallHeight 0.5
}
Wall {
  translation 1.75 1.5 0
  rotation 1 0 0 1.5708
  size 1 0.5 0.1
}
Wall {
  translation 0 1.5 0
  rotation 1 0 0 1.5708
  name "wall(1)"
  size 1.5 0.5 0.1
}
Wall {
  translation 0.75 2 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(2)"
  size 1 0.5 0.1
}
Wall {
  translation -1.75 1.5 0
  rotation 1 0 0 1.5708
  name "wall(3)"
  size 1 0.5 0.1
}
Wall {
  translation -1.25 2 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(4)"
  size 1 0.5 0.1
}
Wall {
  translation 0 2.5 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(5)"
  size 1 0.5 0.1
}
Wall {
  translation 1.25 0 0
  rotation 1 0 0 1.5708
  name "wall(6)"
  size 2 0.5 0.1
}
Wall {
  translation -0.75 0.75 0
  rotation 1 0 0 1.5708
  name "wall(7)"
  size 2 0.5 0.1
}
Wall {
  translation 0.25 -0.25 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(8)"
  size 2 0.5 0.1
}
Wall {
  translation -1.75 -0.25 0
  rotation 1 0 0 1.5708
  name "wall(9)"
  size 1 0.5 0.1
}
Wall {
  translation 1.25 -0.75 0
  rotation 1 0 0 1.5708
  name "wall(10)"
  size 1 0.5 0.1
}
Wall {
  translation -0.5 -1.25 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(11)"
  size 2 0.5 0.1
}
Wall {
  translation 1.775 -1.5 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(12)"
  size 1.5 0.5 0.1
}
Wall {
  translation 1 -1.75 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(13)"
  size 1 0.5 0.1
}
Wall {
  translation 0.225 -2.25 0
  rotation 1 0 0 1.5708
  name "wall(14)"
  size 1.5 0.5 0.1
}
Wall {
  translation -1 -1 0
  rotation 1 0 0 1.5708
  name "wall(15)"
  size 1 0.5 0.1
}
Wall {
  translation -1.5 -1.5 0
  rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
  name "wall(16)"
  size 1 0.5 0.1
}
Robot {
  translation 1.8 2 0.038242
  rotation 0.0014088084794716342 -0.0004006001433769858 -0.9999989273885214 -0.5529233049278005
  children [
    Lidar {
      translation 0 0 0.06
      rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0943999999999994
      numberOfLayers 1
      maxRange 100
      type "rotating"
    }
    GPS {
    }
    GPS {
      translation 0.05 0 0
      name "head_gps"
    }
    InertialUnit {
      rotation 0 0 1 1.5708
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.135994549596158
        axis 0 1 0
        anchor 0.06 0.07 0
      }
      device [
        RotationalMotor {
          name "motor1"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.06000000000000011 0.07000000000000028 2.544670118079281e-16
        rotation 0 -1 0 1.1359945495961579
        children [
          DEF WHEEL Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.1359954840243558
        axis 0 1 0
        anchor -0.06 0.07 0
      }
      device [
        RotationalMotor {
          name "motor2"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.05999999999999993 0.07000000000000028 1.5139111059946813e-16
        rotation 0 -1 0 1.135995484024356
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1.7759954814970105
        axis 0 1 0
        anchor 0.06 -0.07 0
      }
      device [
        RotationalMotor {
          name "motor3"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation 0.05999999999999999 -0.06999999999999984 -5.5256525854599433e-17
        rotation 0 1 0 1.7759954814970105
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1.7759945443484844
        axis 0 1 0
        anchor -0.06 -0.07 0
      }
      device [
        RotationalMotor {
          name "motor4"
          maxVelocity 20
        }
      ]
      endPoint Solid {
        translation -0.060000000000000026 -0.06999999999999984 -1.643253893187146e-16
        rotation 0 0.9999999999999999 0 1.7759945443484841
        children [
          Shape {
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "my_controller"
}
